### ISSUE TITLE
Amalgamation with avx2 is broken

### ISSUE THREAD

**Issue by dsvi** [randombit/botan#1855]
As of 2.9.0
Config parameters:
`--amalgamation --minimized-build --single-amalgamation-file --enable-modules=chacha20poly1305,chacha_avx2,sha3`
gcc with `-mavx2` says:
```
botan_all.cpp: In member function 'void Botan::ChaCha::chacha_avx2_x8(uint8_t*, uint32_t*, std::size_t)':
botan_all.cpp:1016:4: error: 'SIMD_8x32' has not been declared
```

**Comment by randombit** [comment]
Works for me. Are you sure you are not using 2.8? There was a bug (#1700) in 2.8 that broke in exactly this configuration.

**Comment by randombit** [comment]
Oh sorry, it looks like master works but 2.9 release doesn't either. But there is nothing in the release notes about it. Let me track down the commit that fixed this.

**Comment by randombit** [comment]
Ah, found it. This was a bug specific to Chacha-AVX2 fixed in b8199a7a4

You can work around by adding `simd_avx2` to the list of enabled modules.

**Comment by dsvi** [comment]
oh ok, thanks! with `simd_avx2` it works fine
